Template that passes `cfn-lint` fails once deployed to CloudFormation

### CloudFormation Lint Version

0.85.2

### What operating system are you using?

Ubuntu

### Describe the bug

`cfn-lint` isn't able to validate `AWS::CloudFront::Distribution` correctly.

I had the following resource passing:

```yaml
  CloudFrontDistribution:
    Type: AWS::CloudFront::Distribution
    Properties:
      DistributionConfig:
        Enabled: true
        Origins:
          - DomainName: !GetAtt S3Bucket.DomainName
            Id: !Sub "${BucketName}"
            OriginAccessControlId: !Ref CloudFrontOriginAccessControl
            S3OriginConfig: {}
        DefaultCacheBehavior:
          TargetOriginId: !Sub "${BucketName}"
          ViewerProtocolPolicy: "redirect-to-https"
```

This is not valid though and the deploy would fail:

```sh
[ERROR] ResourceType=AWS::CloudFront::Distribution ResourceName=CloudFrontDistribution ResourceStatus=CREATE_FAILED StatusReason="Resource handler returned message: "Invalid request provided: AWS::CloudFront::Distribution: The parameter ForwardedValues is required.
```

In order for the template to be valid, you either have to provide `ForwardedValues` (deprecated) or a `CachePolicyId`.

### Expected behavior

`cfn-lint` warns me that there's a missing required field.

### Reproduction template

See above.

--- a/src/cfnlint/data/schemas/patches/extensions/all/aws_cloudfront_distribution/manual.json
+++ b/src/cfnlint/data/schemas/patches/extensions/all/aws_cloudfront_distribution/manual.json
@@ -10,6 +10,22 @@
  },
  {
   "op": "add",
+  "path": "/definitions/CacheBehavior/requiredXor",
+  "value": [
+   "ForwardedValues",
+   "CachePolicyId"
+  ]
+ },
+ {
+  "op": "add",
+  "path": "/definitions/DefaultCacheBehavior/requiredXor",
+  "value": [
+   "ForwardedValues",
+   "CachePolicyId"
+  ]
+ },
+ {
+  "op": "add",
   "path": "/definitions/Origin/dependentExcluded",
   "value": {
    "CustomOriginConfig": [

